client/sites-dashboard/components/sites-dashboard.tsx

```diff
@@ -102,28 +102,32 @@ export function SitesDashboard( { queryParams: { search, status = 'all' } }: Sit
 			</PageHeader>
 			<PageBodyWrapper>
 				<>
-					<FilterBar>
-						<SitesSearch
-							searchIcon={ <SitesSearchIcon /> }
-							onSearch={ ( term ) => handleQueryParamChange( 'search', term?.trim() ) }
-							isReskinned
-							placeholder={ __( 'Search by name or domain…' ) }
-							disableAutocorrect={ true }
-							defaultValue={ search }
-						/>
-						<SelectDropdown selectedText={ selectedStatus.title }>
-							{ statuses.map( ( { name, title, count } ) => (
-								<SelectDropdown.Item
-									key={ name }
-									selected={ name === selectedStatus.name }
-									count={ count }
-									onClick={ () => handleQueryParamChange( 'status', 'all' !== name ? name : '' ) }
-								>
-									{ title }
-								</SelectDropdown.Item>
-							) ) }
-						</SelectDropdown>
-					</FilterBar>
+					{ allSites.length > 0 ? (
+						<FilterBar>
+							<SitesSearch
+								searchIcon={ <SitesSearchIcon /> }
+								onSearch={ ( term ) => handleQueryParamChange( 'search', term?.trim() ) }
+								isReskinned
+								placeholder={ __( 'Search by name or domain…' ) }
+								disableAutocorrect={ true }
+								defaultValue={ search }
+							/>
+							<SelectDropdown selectedText={ selectedStatus.title }>
+								{ statuses.map( ( { name, title, count } ) => (
+									<SelectDropdown.Item
+										key={ name }
+										selected={ name === selectedStatus.name }
+										count={ count }
+										onClick={ () => handleQueryParamChange( 'status', 'all' !== name ? name : '' ) }
+									>
+										{ title }
+									</SelectDropdown.Item>
+								) ) }
+							</SelectDropdown>
+						</FilterBar>
```

Comment:
How about extracting this chunk into a component and just referencing it here? It would help us by adding more modularity and separation of concerns. The goal of the `SitesDashboard` component is to specify the sections we want to render, not to include them here. It makes sense from a maintenance perspective because if we want to add more things to that section on the page, we might end up bloating the `SitesDashboard` component itself 👍

Reply:
> How about extracting this chunk into a component and just referencing it here?

@zaguiini Unnecessary complexity at this point, in my opinion. We can solve future problems when they exist.